evap/staff/views.py

```diff
@@ -144,159 +149,194 @@ def __init__(self):
-                evaluations = evaluations_ready_for_evaluation
-                messages.warning(request, ungettext("%(evaluations)d evaluation can not be started, because it was not approved, was already evaluated or its evaluation end date lies in the past. It was removed from the selection.",
-                    "%(evaluations)d evaluations can not be started, because they were not approved, were already evaluated or their evaluation end dates lie in the past. They were removed from the selection.",
-                    difference) % {'evaluations': difference})
-            email_template = EmailTemplate.objects.get(name=EmailTemplate.EVALUATION_STARTED)
-            confirmation_message = _("Do you want to immediately start the following evaluations?")
-
-        elif target_state == 'reviewed':
-            unpublishable_evaluations = [evaluation for evaluation in evaluations if evaluation.state == 'published']
-            difference = len(evaluations) - len(unpublishable_evaluations)
-            if difference:
-                evaluations = unpublishable_evaluations
-                messages.warning(request, ungettext("%(evaluations)d evaluation can not be unpublished, because it's results have not been published. It was removed from the selection.",
-                    "%(evaluations)d evaluations can not be unpublished because their results have not been published. They were removed from the selection.",
-                    difference) % {'evaluations': difference})
-            confirmation_message = _("Do you want to unpublish the following evaluations?")
-
-        elif target_state == 'published':
-            publishable_evaluations = [evaluation for evaluation in evaluations if evaluation.state == 'reviewed']
-            difference = len(evaluations) - len(publishable_evaluations)
-            if difference:
-                evaluations = publishable_evaluations
-                messages.warning(request, ungettext("%(evaluations)d evaluation can not be published, because it's not finished or not all of its text answers have been reviewed. It was removed from the selection.",
-                    "%(evaluations)d evaluations can not be published, because they are not finished or not all of their text answers have been reviewed. They were removed from the selection.",
-                    difference) % {'evaluations': difference})
-            email_template = EmailTemplate.objects.get(name=EmailTemplate.PUBLISHING_NOTICE)
-            confirmation_message = _("Do you want to publish the following evaluations?")
-
-    if not evaluations:
+    applicable_evaluations = list(filter(operation.applicable_to, evaluations))
+    difference = len(evaluations) - len(applicable_evaluations)
+    if difference:
+        evaluations = applicable_evaluations
+        messages.warning(request, operation.warning_for_inapplicables(difference))
+    if not evaluations:  # no evaluations where applicable or none were selected
```

Comment:
i think this should be `applicable_evaluations`, could you check?

Reply:
Should be correct, as `evaluations` points to `applicable_evaluations` if there was a difference between the two. `evaluations` is also used in the template context. 

A way to make this clearer is to have `evaluations` be the filtered list from the beginning, and keeping a `selected_evaluations` list for the difference check. What do you think about that?

Reply:
i think `evaluations` and `applicable_evaluations` are fine, i would simply drop the re-purposing two lines above, use `applicable_evaluations` here, and do `evaluations=applicable_evaluations` for the template.